app/service/planning_svc.py

```diff
@@ -272,7 +272,11 @@ async def sort_links(links):
         :return: Sorted links
         :rtype: list(Link)
         """
-        return sorted(links, key=lambda k: (-k.score))
+        normals = [x for x in links if not x.ability.repeatable]
+        repeats = [x for x in links if x.ability.repeatable]  # split by link type, sort, then recombine
+        normals = sorted(normals, key=lambda k: (-k.score))
+        repeats = sorted(repeats, key=lambda k: (-k.score))
+        return normals + repeats
```

Comment:
why the split here?

Reply:
This is an means of addressing a priority problem with repeatable abilities, specifically for the atomic planner. All this is doing is putting the repeat abilities to the end of the list, so we can't end up in a situation where we get stuck running a single command forever with other things left to run. This doesn't affect anything that filters by score, but it will affect anything that filters exclusively by order. 

Technically, this isn't directly related to the Object service base, but since it ties into services and I had it sitting around from other work, I thought I'd put it in now.

Reply:
I've removed this - if the problem resurfaces in the future, we can revisit this potential issue, but right now it's not super pressing.